File: fixtures/hub-flagship-java/src/FlagshipController.java
package hub;

import java.util.Map;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

/**
 * hub-flagship-java — 20-route Spring mirror of hub-flagship-express / go / python / csharp.
 * No invented product UI (D6447). Bodies use ResponseEntity / Map.of / scalar idioms the hub
 * Java→WebIR lift understands (brace-bounded methods + status+body).
 */
@RestController
public class FlagshipController {

  @GetMapping("/health")
  public boolean health() {
    return true;
  }

  @GetMapping("/ping")
  public int ping() {
    return 42;
  }

  @GetMapping("/version")
  public int version() {
    return 1;
  }

  @GetMapping("/ready")
  public String ready() {
    return "ok";
  }

  @GetMapping("/count")
  public int count() {
    return 3;
  }

  @GetMapping("/flag")
  public String flag() {
    return "chrysalis";
  }

  @GetMapping("/build")
  public int build() {
    return 2026;
  }

  @GetMapping("/tier")
  public String tier() {
    return "gold";
  }

  @GetMapping("/meta")
  public Map<String, Object> meta() {
    return Map.of("service", "hub-flagship-java", "version", 1);
  }

  @PostMapping("/echo")
  public Map<String, Object> echo() {
    return Map.of("echo", true);
  }

  @GetMapping("/items")
  public boolean items() {
    return true;
  }

  @GetMapping("/items/{id}")
  public Map<String, Object> getItem(@PathVariable String id) {
    return Map.of("id", id);
  }

  @PostMapping("/items")
  public ResponseEntity<Map<String, Object>> createItem() {
    return ResponseEntity.status(201).body(Map.of("created", true));
  }

  @GetMapping("/search")
  public Map<String, Object> search(@RequestParam(name = "q", defaultValue = "") String q) {
    return Map.of("q", q);
  }

  @PutMapping("/items/{id}")
  public Map<String, Object> putItem(@PathVariable String id) {
    return Map.of("updated", true, "id", id);
  }

  @DeleteMapping("/items/{id}")
  public boolean deleteItem(@PathVariable String id) {
    return true;
  }

  @PatchMapping("/items/{id}")
  public Map<String, Object> patchItem(@PathVariable String id) {
    return Map.of("patched", true, "id", id);
  }

  @GetMapping("/users/{userId}")
  public String getUser(@PathVariable String userId) {
    return userId;
  }

  @GetMapping("/stats")
  public int stats() {
    return 3;
  }

  @PostMapping("/notify")
  public ResponseEntity<Map<String, Object>> notify() {
    return ResponseEntity.status(202).body(Map.of("ok", true));
  }
}

File: fixtures/hub-flagship-kotlin/src/FlagshipController.kt
package hub

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

/**
 * hub-flagship-kotlin — 20-route Spring Kotlin mirror of hub-flagship-express / java / go.
 * No invented product UI (D6447). Bodies use mapOf / ResponseEntity / scalar idioms the hub
 * Kotlin→WebIR lift understands (brace + expression fun bodies + path/query refs).
 */
@RestController
class FlagshipController {

  @GetMapping("/health")
  fun health(): Boolean = true

  @GetMapping("/ping")
  fun ping(): Int = 42

  @GetMapping("/version")
  fun version(): Int = 1

  @GetMapping("/ready")
  fun ready(): String = "ok"

  @GetMapping("/count")
  fun count(): Int = 3

  @GetMapping("/flag")
  fun flag(): String = "chrysalis"

  @GetMapping("/build")
  fun build(): Int = 2026

  @GetMapping("/tier")
  fun tier(): String = "gold"

  @GetMapping("/meta")
  fun meta(): Map<String, Any> = mapOf("service" to "hub-flagship-kotlin", "version" to 1)

  @PostMapping("/echo")
  fun echo(): Map<String, Any> = mapOf("echo" to true)

  @GetMapping("/items")
  fun items(): Boolean = true

  @GetMapping("/items/{id}")
  fun getItem(@PathVariable id: String): Map<String, Any> = mapOf("id" to id)

  @PostMapping("/items")
  fun createItem(): ResponseEntity<Map<String, Any>> =
    ResponseEntity.status(201).body(mapOf("created" to true))

  @GetMapping("/search")
  fun search(@RequestParam(name = "q", defaultValue = "") q: String): Map<String, Any> =
    mapOf("q" to q)

  @PutMapping("/items/{id}")
  fun putItem(@PathVariable id: String): Map<String, Any> = mapOf("updated" to true, "id" to id)

  @DeleteMapping("/items/{id}")
  fun deleteItem(@PathVariable id: String): Boolean = true

  @PatchMapping("/items/{id}")
  fun patchItem(@PathVariable id: String): Map<String, Any> = mapOf("patched" to true, "id" to id)

  @GetMapping("/users/{userId}")
  fun getUser(@PathVariable userId: String): String = userId

  @GetMapping("/stats")
  fun stats(): Int = 3

  @PostMapping("/notify")
  fun notify(): ResponseEntity<Map<String, Any>> =
    ResponseEntity.status(202).body(mapOf("ok" to true))
}
